lib/api/model/model.dart

```diff
@@ -118,6 +118,50 @@ class User {
   Map<String, dynamic> toJson() => _$UserToJson(this);
 }
 
+/// As in `streams` in the initial snapshot.
+///
+/// Not called `Stream` because dart:async uses that name.
+@JsonSerializable(fieldRename: FieldRename.snake)
+class ZulipStream {
+  final int streamId;
+  final String name;
+  final String description;
+  final String renderedDescription;
+
+  final int dateCreated;
+  final int? firstMessageId;
+
+  final bool inviteOnly;
+  final bool isWebPublic;
```

Comment:
In `Subscription` we say this:

```dart
final bool? isWebPublic; // TODO(server-??): doc doesn't say when added
```

and the API doc is the same for this property on `streams[]` and on `subscriptions[]`:

> Whether the stream has been configured to allow unauthenticated access to its message history from the web.

Do we think the same TODO might apply here?

Reply:
Hmm, good catch, thanks.

In addition to the TODO, there's the making it optional — so in `Subscription.isWebPublic` it looks like we figured it may be newer than the oldest server version we're supporting.

However, for stream objects there is a mention in the overall API changelog, back in server-2.1:
https://zulip.com/api/changelog#changes-in-zulip-21
> Added `is_web_public` field to Stream objects. This field is intended to support web-public streams.

That's back in the early period of the changelog and the API docs, when we weren't as systematic about having a changelog entry and a "Changes:" note for each API change.

So I think the conclusion is that it's always present on streams for the servers we support. Which means the same is *probably* true of subscriptions, but it's not documented.

Reply:
(I'll add a comment)